### PR TITLE
Magnus/faults display

### PR DESCRIPTION
**Pull request by mglyngberg**
Replaced motor fault icons with fault labels for race mode

### CODE REVIEW

File: src/ViewLayer/RaceModeDisplay/RaceModeFaults/FaultList.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+              , FaultLabel(IPM_MOTOR_TEMPERATURE_LIMIT, YELLOW, 1, false)
+             })
+{
+}
+
+FaultList::~FaultList()
+{
+}
+
+FaultLabel FaultList::getHighestActivePriorityLabel() const
```
Comment (bill-luu) [inline src/ViewLayer/RaceModeDisplay/RaceModeFaults/FaultList.cpp]: This function should just return the highest active priority, and then another function `setHighestActivePriorityLabel` can have the logic for determining it.

File: src/ViewLayer/RaceModeDisplay/RaceModeFaults/FaultList.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    FaultLabel getHighestActivePriorityLabel() const;
+    QVector<FaultLabel> getErrorLabelList() const;
+    QVector<FaultLabel> getLimitLabelList() const;
+    void updateErrors(const ErrorFlags& errorFlags);
+    void updateLimits(const LimitFlags& limitFlags);
+
+private:
+
+    QVector<FaultLabel> errorList_;
```
Comment (bill-luu) [inline src/ViewLayer/RaceModeDisplay/RaceModeFaults/FaultList.h]: Since this is a vector, it's a little bit misleading to call this a `errorList`.

**Comment by mglyngberg** [comment]
There are some issues with battery faults either in the dashboard or the test-tool: 
- In racemode the "Internal Communication Fault" flag is not updated
- The debug display does not display all of the battery faults and some are displayed when they shouldn't
- None of the limit flags are updated

File: src/ViewLayer/Faults/FaultLabel/FaultLabel.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    FaultLabel(const FaultLabel& faultLabel);
+
+    ~FaultLabel();
+
+    QString name() const;
+    QColor color() const;
+    int priority() const;
+    bool isActive() const;
+
+    void setIsActive(bool value);
```
Comment (bill-luu) [inline src/ViewLayer/Faults/FaultLabel/FaultLabel.h]: Convention would be `setActive`

File: src/ViewLayer/Faults/MotorFaults/MotorFaultList.cpp [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    FaultLabel(MOTOR_OVER_SPEED, RED, 0, false)
+    , FaultLabel(SOFTWARE_OVER_CURRENT, RED, 0, false)
+    , FaultLabel(DC_BUS_OVER_VOLTAGE, RED, 0, false)
+    , FaultLabel(BAD_MOTOR_POSITION_HALL_SEQUENCE, RED, 0, false)
+    , FaultLabel(WATCHDOG_CAUSED_LAST_RESET, RED, 0, false)
+    , FaultLabel(CONFIG_READ_ERROR, RED, 0, false)
+    , FaultLabel(RAIL_UNDER_VOLTAGE_LOCK_OUT, RED, 0, false)
+    , FaultLabel(DESATURATION_FAULT, RED, 0, false)
+})
+, limitLabels_({FaultLabel(OUTPUT_VOLTAGE_PWM_LIMIT, YELLOW, 1, false)
```
Comment (bill-luu) [inline src/ViewLayer/Faults/MotorFaults/MotorFaultList.cpp]: Let's format it the way you did the error labels.
````
limitLabels_(
    {
       .....
    }
)

File: src/config.ini [hunk 5]
```diff
@@ -4,7 +4,7 @@
 packetTitle=Gen5
 
 [SourceAddress]
-ipAddress=138.197.78.235
+ipAddress=127.0.0.1.
```
Comment (bill-luu) [inline src/config.ini]: Let's keep this as 127.0.0.1

File: src/ViewLayer/Faults/FaultLabel/FaultLabel.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+public:
+    FaultLabel();
+    FaultLabel(QString, FaultPriorities::PRIORITY, bool);
+    FaultLabel(const FaultLabel&);
+
+    ~FaultLabel();
+
+    QString name() const;
+    QColor color() const;
+    int priority() const;
```
Comment (bill-luu) [inline src/ViewLayer/Faults/FaultLabel/FaultLabel.h]: Instead of int, it should be the enum we defined.

**Pull request merged**